api/src/main/resources/_core/database/mysql/z/obs/sp_mamba_z_encounter_obs_update.sql

```diff
@@ -1,38 +1,65 @@
--- $BEGIN
-
-CREATE TEMPORARY TABLE mamba_temp_value_coded_values
-    CHARSET = UTF8MB4 AS
-SELECT m.concept_id AS concept_id,
-       m.uuid       AS concept_uuid,
-       m.name       AS concept_name
-FROM mamba_dim_concept m
-WHERE concept_id in (SELECT DISTINCT obs_value_coded
-                     FROM mamba_z_encounter_obs
-                     WHERE obs_value_coded IS NOT NULL);
-CREATE INDEX mamba_idx_concept_id ON mamba_temp_value_coded_values (concept_id);
-
--- update obs_value_coded (UUIDs & Concept value names)
-UPDATE mamba_z_encounter_obs z
-    INNER JOIN mamba_temp_value_coded_values mtv
-    ON z.obs_value_coded = mtv.concept_id
-SET z.obs_value_text       = mtv.concept_name,
-    z.obs_value_coded_uuid = mtv.concept_uuid
-WHERE z.obs_value_coded IS NOT NULL;
+DROP PROCEDURE IF EXISTS sp_mamba_z_encounter_obs_update;
+
+DELIMITER //
+
+CREATE PROCEDURE sp_mamba_z_encounter_obs_update()
+BEGIN
+    DECLARE total_records INT;
+    DECLARE batch_size INT DEFAULT 100000; -- batch size
+    DECLARE mamba_offset INT DEFAULT 0;
+
+    CREATE TEMPORARY TABLE mamba_temp_value_coded_values
+        CHARSET = UTF8MB4 AS
+    SELECT m.concept_id AS concept_id,
+           m.uuid       AS concept_uuid,
+           m.name       AS concept_name
+    FROM mamba_dim_concept m
+    WHERE concept_id in (SELECT DISTINCT obs_value_coded
+                         FROM mamba_z_encounter_obs
+                         WHERE obs_value_coded IS NOT NULL);
+    CREATE INDEX mamba_idx_concept_id ON mamba_temp_value_coded_values (concept_id);
+
+    SELECT COUNT(*)
+    INTO total_records
+    FROM mamba_z_encounter_obs z
+             INNER JOIN mamba_temp_value_coded_values mtv
+                        ON z.obs_value_coded = mtv.concept_id
+    WHERE z.obs_value_coded IS NOT NULL;
+
+    WHILE mamba_offset < total_records
+        DO
+            -- update obs_value_coded (UUIDs & Concept value names)
+            SET @sql = CONCAT('UPDATE mamba_z_encounter_obs z
+                INNER JOIN mamba_temp_value_coded_values mtv
+                ON z.obs_value_coded = mtv.concept_id
+                    SET z.obs_value_text = mtv.concept_name,
+                    z.obs_value_coded_uuid = mtv.concept_uuid
+                WHERE z.obs_value_coded IS NOT NULL
+                LIMIT ', batch_size, ' OFFSET ', mamba_offset);
```

Comment:
@smallgod  I am running into this error [2024-10-22 09:07:24] [42000][1064] You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'OFFSET 0' at line 7